### PR TITLE
add lagrange method

### PR DESCRIPTION
**Pull request by yahya-mohammed07**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: numerical_methods/lagrange.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include <iostream>
```
Comment (realstealthninja) [inline numerical_methods/lagrange.cpp:1]: Document headers

File: numerical_methods/lagrange.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    lagrange += temp;
+  }
+  return lagrange;
+}
+
+int main() {
+  std::vector<float> xi{1, 2, 3, 4, 5};
+  std::vector<float> yi{2, 3, 4, 6, 7};
+  std::cout << lagrange(xi, yi, 3.);
+}
```
Comment (realstealthninja) [inline numerical_methods/lagrange.cpp:30]:
```suggestion
}

```

File: numerical_methods/lagrange.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    for (size_t j{}; j < xi.size(); ++j) {
+      if (i != j) {
+        temp *= (x - xi[j]) / (xi[i] - xi[j]);
+      }
+    }
+    lagrange += temp;
+  }
+  return lagrange;
+}
+
```
Comment (realstealthninja) [inline numerical_methods/lagrange.cpp:25]: Should document main as well as add a seperate function for testing

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: numerical_methods/lagrange.cpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  constexpr std::array<float, 5> xi{1, 2, 3, 4, 5};
+  constexpr std::array<float, 5> yi{2, 3, 4, 6, 7};
+  // should return true
+  return (lagrange(xi, yi, 3.) == 4.0f);
+};
+
+int main() {
+  // shows the the result for the calculations
+  std::cout << std::boolalpha << check();
+}
```
Comment (realstealthninja) [inline numerical_methods/lagrange.cpp:37]:
```suggestion
}

```

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: numerical_methods/lagrange.cpp (hunk 4 above)
Comment (yahya-mohammed07) [reply numerical_methods/lagrange.cpp:37]: I'm sorry, but i don't quite understand what changes you request?

**Comment by yahya-mohammed07** [comment]
@realstealthninja  can u explain the last request?

**Pull request closed without merging**